#! /bin/sh /usr/share/dpatch/dpatch-run
## 33_avoid_memory_overrun.dpatch by <mathieu.malaterre@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Dont apply the -s strip flag to the shared lib debian will do it for us 
## DP: where necessary
## DP: Remove unnecessary linkage on lstdc++ and lm and add missing linkage for -lm

@DPATCH@

Index: ./libopenjpeg/tcd.c
===================================================================
--- ./libopenjpeg/tcd.c
+++ ./libopenjpeg/tcd.c
@@ -332,7 +332,7 @@
 							cblk->y0 = int_max(cblkystart, prc->y0);
 							cblk->x1 = int_min(cblkxend, prc->x1);
 							cblk->y1 = int_min(cblkyend, prc->y1);
-							cblk->data = (unsigned char*) opj_calloc(8192+2, sizeof(unsigned char));
+							cblk->data = (unsigned char*) opj_calloc(9728+2, sizeof(unsigned char));
 							/* FIXME: mqc_init_enc and mqc_byteout underrun the buffer if we don't do this. Why? */
 							cblk->data += 2;
 							cblk->layers = (opj_tcd_layer_t*) opj_calloc(100, sizeof(opj_tcd_layer_t));
